homeassistant/components/obihai/button.py

```diff
@@ -0,0 +1,59 @@
+"""Obihai button module."""
+
+from __future__ import annotations
+
+from pyobihai import PyObihai
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import entity_platform
+
+from .connectivity import ObihaiConnection
+from .const import OBIHAI
+
+BUTTON_DESCRIPTION = ButtonEntityDescription(
+    key="reboot",
+    name=f"{OBIHAI} Reboot",
+    device_class=ButtonDeviceClass.RESTART,
+    entity_category=EntityCategory.CONFIG,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: entity_platform.AddEntitiesCallback,
+) -> None:
+    """Set up the Obihai sensor entries."""
+    username = entry.data[CONF_USERNAME]
+    password = entry.data[CONF_PASSWORD]
+    host = entry.data[CONF_HOST]
+    requester = ObihaiConnection(host, username, password)
```

Comment:
This raises a red flag for me, is every platform creating their own instance? Why?

Instead it should be created during entry setup, so the platforms can just use them.